packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts

```diff
@@ -761,7 +818,13 @@ export class AppSyncModelJavaVisitor<
   }
 
   protected generateModelFieldAnnotation(field: CodeGenField): string {
-    const annotationArgs: string[] = [`targetType="${field.type}"`, !field.isNullable ? 'isRequired = true' : ''].filter(arg => arg);
+    const authDirectives: AuthDirective[] = (field.directives.filter(d => d.name === 'auth') as any) as AuthDirective[];
+    const authRules = this.generateAuthRules(authDirectives);
+    const annotationArgs: string[] = [
+      `targetType="${field.type}"`,
+      !field.isNullable ? 'isRequired = true' : '',
+      authRules.length ? `authRules = ${authRules}` : '',
+    ].filter(arg => arg);
```

Comment:
``` javascript
.filter(arg => arg);
```
this condition will return same array right?

Reply:
Yup will remove this filter

Reply:
This code was already present, I believe it is there for formatting purposes (checking for string type)